src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Compatibility.Common.targets

```diff
@@ -45,32 +45,23 @@
       ReferencePaths="@(PackageValidationReferencePath)" />
   </Target>
 
-  <PropertyGroup>
-    <_GetReferencePathFromInnerProjectsDependsOn Condition="'$(IsCrossTargetingBuild)' != 'true'">_GetReferencePathForPackageValidation</_GetReferencePathFromInnerProjectsDependsOn>
-    <_GetReferencePathFromInnerProjectsDependsOn Condition="'$(IsCrossTargetingBuild)' == 'true'">_ComputeTargetFrameworkItems</_GetReferencePathFromInnerProjectsDependsOn>
-  </PropertyGroup>
-
   <Target Name="_GetReferencePathForPackageValidation"
           DependsOnTargets="ResolveReferences"
           Returns="@(_ReferencePathWithTargetFramework)">
     <ItemGroup>
-      <_ReferencePathWithTargetFramework Include="@(ReferencePath)"  TargetFramework="$(TargetFramework)" />
+      <_ReferencePathWithTargetFramework Include="@(ReferencePath)" TargetFramework="$(TargetFramework)" />
     </ItemGroup>
   </Target>
 
+  <!-- Depends on NuGet's _GetTargetFrameworksOutput target to calculate inner target frameworks. -->
   <Target Name="_GetReferencePathFromInnerProjects"
-          DependsOnTargets="$(_GetReferencePathFromInnerProjectsDependsOn)"
+          DependsOnTargets="_GetTargetFrameworksOutput"
```

Comment:
Previously we used `_ComputeTargetFrameworkItems` which, though private, has a lot of other dependencies in this repo.  Why did we need to change to `_GetTargetFrameworksOutput` and is the NuGet team OK with that?

Reply:
Even though we depend on a NuGet internal here, we should be able to catch any change to it by adding a test that utilizes this code path. NuGet directly inserts into dotnet/sdk so that would be caught as part of the insertion. Of course we want to keep the noise low but I wouldn't expect this to change often.

> Why did we need to change to _GetTargetFrameworksOutput

As discussed offline, we intentionally want to walk the same code path as NuGet does during packaging to keep our logic in sync. The change that requires this target is the MSBuild task that now batches on the `TargetFrameworks` property.

> and is the NuGet team OK with that?

@dotnet/nuget-team are you ok with us relying on this private `_GetTargetFrameworksOutput` target?

Reply:
This functionality ships with every new SDK version right? There's a world where someone installs a higher version of the nuget.build.tasks.pack package that has tweaked that target and it might break you.

We haven't changed that target in 5 years, so I don't expect a lot of movement there either. 
The standard `"internal"` target disclosure applies, but it should be reasonably easy react to any changes.

Reply:
> This functionality ships with every new SDK version right?

Yes and no. This does ship inside the SDK, but it also ships as a standalone package that someone can reference in order to pick a different version of this functionality (similar to how the Roslyn compiler ships in the SDK but you can also opt to use a different version). I'm assuming that fact doesn't change much your stance given that as you point out, this target hasn't been changed in 5 years?

Reply:
IMO we don't want to advertise being able to reference a new version of the Compatibility tooling via a Package Reference. Roslyn or the linker who ship these toolset packages don't advertise them either as they are only meant for internal consumption of our core stack repositories.

Reply:
I wasn't suggesting advertising them. My point was more that because we have that shipping vehicle, it does mean that we could run into additional cases of SDK and PackageValidation comming from two different builds/branches/releases, so I was wondering if that might aggravate @nkolev92 concern above since he specifically asked if the shipping vehicle for this was only the SDK.

Reply:
What I find interesting is that the `Microsoft.DotNet.Compatibility` package ships publicly. I thought we wanted to treat it as a transport only package but it shipped/ships to nuget.org. We could change that if we would like to.

Reply:
> , given that referencing the package and not using the feature as part of the SDK isn't a supported scenario (beyond experimentation).

Make sure you document that :)